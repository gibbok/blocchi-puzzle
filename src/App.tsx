import * as React from 'react';
import { Provider } from 'react-redux';
import { store } from './store/reducer';
import { ScreenContainer } from './containers/ScreenContainer';
import GlobalStyle from './components/GlobalStyle';
import { detectorBrowser } from '../src/utils';
import { BrowserNotSupported } from './components';
import { detect } from 'detect-browser';
import { AppContextType, AppContextProvider } from './context';
import { BrowserInfo } from './game/types';

export function App(): JSX.Element {
  const browser = detect();
  const browserInfo: BrowserInfo | undefined =
    browser && browser.name && browser.version
      ? { name: browser.name, version: browser.version }
      : undefined;
  const isValidBrowser = detectorBrowser(browserInfo);

  const handleSetRepeat = (repeat: boolean) =>
    setAppState((s) => ({
      ...s,
      repeat,
    }));

  const defaultContext: AppContextType = {
    browserInfo: browserInfo,
    repeat: false,
    setRepeat: handleSetRepeat,
  };

  const [appState, setAppState] = React.useState<AppContextType>(defaultContext);

  return (
    <AppContextProvider value={appState}>
      <Provider store={store}>
        <GlobalStyle />
        {isValidBrowser ? <ScreenContainer /> : <BrowserNotSupported />}
      </Provider>
    </AppContextProvider>
  );
}
