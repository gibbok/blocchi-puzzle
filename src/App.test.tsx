import React from 'react';
import { App } from './App';
import renderer from 'react-test-renderer';

describe('<App />', () => {
  it('should render', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
