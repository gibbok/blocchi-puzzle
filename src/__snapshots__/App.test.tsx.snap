// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<div
  className="sc-fzolEj bilGol"
>
  <div
    className="sc-fzqBkg dMvPDQ"
  >
    Browser not supported! ðŸ˜±
  </div>
  <div
    className="sc-fzqPZZ iMyiKw"
  >
    Please use the latest Chrome or Firefox.
  </div>
</div>
`;
